src/framer_async.rs

```diff
@@ -72,45 +76,67 @@ where
         stream.send(tx_buf).await.map_err(FramerError::Io)?;
         stream.flush().await.map_err(FramerError::Io)?;
 
-        loop {
-            match stream.next().await {
-                Some(buf) => {
-                    let buf = buf.map_err(FramerError::Io)?;
-                    let buf = buf.as_ref();
-
-                    match self.websocket.client_accept(&web_socket_key, buf) {
-                        Ok((len, sub_protocol)) => {
-                            // "consume" the HTTP header that we have read from the stream
-                            // read_cursor would be 0 if we exactly read the HTTP header from the stream and nothing else
-
-                            // copy the remaining bytes to the end of the rx_buf (which is also the end of the buffer) because they are the contents of the next websocket frame(s)
-                            let from = len;
-                            let to = buf.len();
-                            let remaining_len = to - from;
-
-                            if remaining_len > 0 {
-                                let rx_start = rx_buf.len() - remaining_len;
-                                rx_buf[rx_start..].copy_from_slice(&buf[from..to]);
-                                self.rx_remainder_len = remaining_len;
-                            }
-
-                            return Ok(sub_protocol);
-                        }
-                        Err(crate::Error::HttpHeaderIncomplete) => {
-                            // TODO: continue reading HTTP header in loop
-                            panic!("oh no");
-                        }
-                        Err(e) => {
-                            return Err(FramerError::WebSocket(e));
+        match stream.next().await {
```

Comment:
What happened to the loop? I believe that we need it if we only receive tiny chunks of data from the stream, perhaps not enough to read everything we need to complete the connect operation.

Reply:
The loop is never used, because we either return or panic inside it. However I now see that you probably wanted to handle the ```HttpHeaderIncomplete``` in the future instead of panicking, so if you want me to I can undo that change.